algosdk/future/transaction.py

```diff
@@ -3008,3 +3008,32 @@ def assign_group_id(txns, address=None):
             tx.group = gid
             result.append(tx)
     return result
+
+def wait_for_confirmation(algod_client, transaction_id, timeout=None, **kwargs):
+    """
+    Block until a pending transaction is confirmed by the network.
+
+    Args:
+        transaction_id (str): transaction ID
+        timeout (int, optional): The number of rounds to block for before
+            exiting with an Exception. If not supplied, there is no timeout.
+    """
+    last_round = algod_client.status()['last-round'] 
+    current_round = last_round + 1
+
+    while True:
+        # Check that the `timeout` has not passed
+        if timeout is not None and current_round > last_round + timeout:
+            raise error.AlgodResponseError(f"Wait for transaction id {transaction_id} timed out")
+
+        tx_info = algod_client.pending_transaction_info(transaction_id, **kwargs)
+
+        # The transaction has been confirmed
+        if 'confirmed-round' in tx_info:
+            return
```

Comment:
It would be useful to return the `tx_info` here. Often there is a need to check the some properties of this after waiting.